docs/gravitee.md

```diff
@@ -0,0 +1,96 @@
+# Developer Portal APIs
+
+The Developer Portal is where the Orchestra Cities APIs are registered, and the
+policies defined. The following section describes how these APIs are registered
+in the Developer Portal and how policies and Oauth2 authorization are handled.
+
+## The Authorization flow
+
+1. The user requests a token for the client API from Keycloak. The token comes
+   from a public client in Keycloak, so no secret is required to obtain a token.
+   However it's important to note that the public clients set up in Keycloak will
+   limit what the given tokens enable users to do, such as user management operations.
+   The user requests a specific scope with the request (e.g. entity:read). A list
+   of scopes is available in the **Keycloak Configuration** section of this documentation.
+
+2. When then token is used, it is validated against Keycloak's confidential client
+   (the resource-server client). The operation being attempted (e.g. creating an
```

Comment:
so the steps are three
1. get token for public client
2. validate token for private client 
3. perform auth based on token content